Removed white background from site icon image view

stevebaranski opened the pull request:
### Description

Addresses an inconsistency where in _My Sites_ (i.e., `BlogListViewController`), site icons do not have a background applied. On the Site Details (`BlogDetailsViewController`), however, a white background is applied to the image. This has an adverse impact on images with transparency.

It appears that the white background was introduced along with a border to make the placeholder site icon look nice. The fix applies the border when the placeholder is in use, and effectively removes it in other cases. In doing so, it improves consistency with both Calypso and the aforementioned in-app blog list.

To test:

- Checkout the branch and confirm that existing tests pass.
- Review the code and make sure everything looks okay.
- Manually test the app with a combination of sites that both include & omit site icons. Also test by adding & removing a site icon.

Update release notes:

- [X] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### Screenshots

For context, the following shows the site list & details screens **before** the change was made.

![context](https://user-images.githubusercontent.com/221062/52925134-8eba0a00-32e4-11e9-85b6-cb8ee5adbf3a.png)

The following image depicts a site icon **before** & **after** the change is applied.

![beforeandafter](https://user-images.githubusercontent.com/221062/52925158-a72a2480-32e4-11e9-9e26-c15ac9a9ced7.png)

stevebaranski commented:
@nheagy would you be willing to take a look at this?

frosty commented on RELEASE-NOTES.txt:
```diff
@@ -1,2 +1,3 @@
 * Added connection error alert in Sharing screen.
 * Increased padding at the bottom of the share extension's editor, to make typing a longer post a bit more comfortable.
+* Removes the white background color applied to the site icon.
```
```suggestion
* Removes the white background color applied to the site icon on the site details screen.
```

frosty commented on WordPress/Classes/Extensions/UIImageView+SiteIcon.swift:
```diff
@@ -66,13 +66,19 @@ extension UIImageView {
+        setImageWith(request, placeholderImage: placeholderImage, success: { [weak self] (_, _, image) in
+            self?.image = image
+            self?.removePlaceholderBorder()
+        }) { [weak self] (_, _, _) in
```
Nitpick, but could we keep the name of the final closure instead of using trailing closure syntax in this case? I think it's a little clearer.

stevebaranski commented:
@frosty thanks for the thoughtful review. 

I had not fully appreciated the breadth of the impact a change to `UIImageView+SiteIcon.swift` would have. As we discussed via Slack, there are a few other places that set borders - some with different colors.

In addition to addressing the comments you've noted here, I pushed an additional commit that improves the consistency with which the border is applied to the placeholder in other cases, including `LoginEpilogueTableViewController`.

I believe this is ready for another look when you have a free moment. Thank you!

stevebaranski commented:
The workflow has failed and is waiting to be enqueued, but each individual phase has passed.
